### PR TITLE
Include runtime framework name as part of DominoInvocation message

### PR DESCRIPTION
**Pull request by almiliMSFT**
Without this, it's is hard to know if a build was running on .NET Core or .NET Framework.

### CODE REVIEW

File: Public/Src/App/Bxl/Tracing/Log.cs [hunk 1]
```diff
@@ -54,7 +54,7 @@ public abstract partial class Logger
             // Prevent this from going to the log. It is only for ETW and telemetry. DominoInvocationForLocalLog is for the log.
             Keywords = (int)Keywords.SelectivelyEnabled,
             Message = AppInvocationMessage)]
-        public abstract void DominoInvocation(LoggingContext context, string commandLine, BuildInfo buildInfo, MachineInfo machineInfo, string sessionIdentifier, string relatedSessionIdentifier, string startupDirectory, string mainConfig);
+        public abstract void DominoInvocation(LoggingContext context, string commandLine, BuildInfo buildInfo, MachineInfo machineInfo, string sessionIdentifier, string relatedSessionIdentifier, string startupDirectory, string mainConfig, bool isDotNetCore);
```
Comment (almiliMSFT) [inline Public/Src/App/Bxl/Tracing/Log.cs]: The comment for this event says
> WDG has Asimov telemetry listening to this event. Any change to an existing field will require a breaking change announcement

Adding a new field should be backward compatible and thus not constitute a breaking change.

File: Public/Src/App/Bxl/BuildXLApp.cs [hunk 2]
```diff
@@ -924,7 +924,9 @@ private AppResult RunWithLoggingScope(Func<PerformanceMeasurement, AppResult> ru
                         translatedLogDirectory,
                         m_configuration.InCloudBuild(),
                         Directory.GetCurrentDirectory(),
-                        m_initialConfiguration.Startup.ConfigFile.ToString(m_pathTable));
+                        m_initialConfiguration.Startup.ConfigFile.ToString(m_pathTable),
+                        OperatingSystemHelper.IsDotNetCore
```
Comment (dannyvv) [inline Public/Src/App/Bxl/BuildXLApp.cs]: This is fine, but did you consider being more specific i.e. "net472" vs "netcoreapp30" vs "netcoreapp31" when it is released?

Comment (almiliMSFT) [reply Public/Src/App/Bxl/BuildXLApp.cs]: that would be better, agreed.

Will change

File: Public/Src/App/Bxl/Tracing/Log.cs (hunk 1 above)
Comment (dannyvv) [reply Public/Src/App/Bxl/Tracing/Log.cs]: 👍

File: Public/Src/Utilities/Utilities/OperatingSystemHelper.cs [hunk 3]
```diff
@@ -241,6 +253,20 @@ public static string GetInstalledDotNetFrameworkVersion()
...
+        /// Returns the current runtime version.
+        /// 
+        /// On .NET Core, the return string is something like ".NETCoreApp, Version=v2.0"
+        /// 
+        /// On .NET Framework, the return string is something like ".NETFramework, Version = v4.7.2".
+        /// </summary>
+        public static string GetRuntimeFrameworkNameAndVersion()
+        {
+            return IsDotNetCore
+                ? Assembly.GetEntryAssembly()?.GetCustomAttribute<TargetFrameworkAttribute>()?.FrameworkName ?? "unknown"
```
Comment (almiliMSFT) [inline Public/Src/Utilities/Utilities/OperatingSystemHelper.cs]:
```suggestion
                ? Assembly.GetEntryAssembly()?.GetCustomAttribute<TargetFrameworkAttribute>()?.FrameworkName ?? ".NETCore"
```

File: Public/Src/Utilities/Utilities/OperatingSystemHelper.cs [hunk 4]
```diff
@@ -241,6 +253,20 @@ public static string GetInstalledDotNetFrameworkVersion()
...
+        /// Returns the current runtime version.
+        /// 
+        /// On .NET Core, the return string is something like ".NETCoreApp, Version=v2.0"
+        /// 
+        /// On .NET Framework, the return string is something like ".NETFramework, Version = v4.7.2".
+        /// </summary>
+        public static string GetRuntimeFrameworkNameAndVersion()
+        {
+            return IsDotNetCore
+                ? Assembly.GetEntryAssembly()?.GetCustomAttribute<TargetFrameworkAttribute>()?.FrameworkName ?? ".NETCore"
```
Comment (almiliMSFT) [inline Public/Src/Utilities/Utilities/OperatingSystemHelper.cs]:
```suggestion
                ? Assembly.GetEntryAssembly()?.GetCustomAttribute<TargetFrameworkAttribute>()?.FrameworkName ?? ".NETCoreApp"
```

File: Public/Src/Utilities/Utilities/OperatingSystemHelper.cs [hunk 5]
```diff
@@ -241,6 +253,20 @@ public static string GetInstalledDotNetFrameworkVersion()
...
         }
 
+        /// <summary>
+        /// Returns the current runtime version.
+        /// 
+        /// On .NET Core, the return string is something like ".NETCoreApp, Version=v2.0"
+        /// 
+        /// On .NET Framework, the return string is something like ".NETFramework, Version = v4.7.2".
+        /// </summary>
+        public static string GetRuntimeFrameworkNameAndVersion()
```
Comment (dannyvv) [inline Public/Src/Utilities/Utilities/OperatingSystemHelper.cs]: Cool, even more generic and future-compatible than I was thinking! :)

**Comment by dannyvv** [comment]
> Thanks for adding, this is going to be super helpful

👍

**Pull request merged**